htlcswitch/interceptable_switch.go

```diff
@@ -137,21 +142,59 @@ func (f *interceptedForward) Resume() error {
 	return f.htlcSwitch.ForwardPackets(f.linkQuit, f.packet)
 }
 
-// Fail forward a failed packet to the switch.
-func (f *interceptedForward) Fail() error {
-	update, err := f.htlcSwitch.cfg.FetchLastChannelUpdate(
-		f.packet.incomingChanID,
-	)
-	if err != nil {
-		return err
+// Fails notifies the intention to fail an existing hold forward with an
+// encrypted failure reason.
+func (f *interceptedForward) Fail(reason []byte) error {
+	obfuscatedReason := f.packet.obfuscator.IntermediateEncrypt(reason)
+
+	return f.resolve(&lnwire.UpdateFailHTLC{
+		Reason: obfuscatedReason,
+	})
+}
+
+// FailWithCode notifies the intention to fail an existing hold forward with the
+// specified failure code.
+func (f *interceptedForward) FailWithCode(code lnwire.FailCode) error {
+	shaOnionBlob := sha256.Sum256(f.htlc.OnionBlob[:])
```

Comment:
nit: not all cases require `OnionSHA256` to be filled, consider moving calculation where required.

Reply:
Made closure